src/Server/TCPHandler.cpp

```diff
@@ -310,10 +311,25 @@ void TCPHandler::runImpl()
             query_context->setReadTaskCallback([this]() -> String
             {
                 std::lock_guard lock(task_callback_mutex);
+
+                if (state.is_cancelled)
```

Comment:
We should not always throw exception when query is cancelled (probably only it was explicitly killed).

Reply:
I just want to immediately return from this callback.